File: java/xor_shift_gen/xor.java
class xor 
{
    public static void main(String args[])
    {
        final int SEED = 1;
        int val = SEED;
        long period = 0;

         do {
            val ^= val << 13;
            val ^= val >>> 17;
            val ^= val << 5;
            if ( (val & 0xffffff00) == 0 )
            {
                System.out.printf("period %d, val %x\n", period, val);
            }
            period++;
        } while (val != SEED);

    System.out.printf("%d\n", period);
    }
}

File: c/xor_shift_gen/xor.c

#include <stdio.h>
#include <stdint.h>
#include <inttypes.h>

int main() {
    const int32_t SEED = 1;
    uint32_t val = SEED;
    int64_t period = 0;

    do {
        val ^= val << 13;
        val ^= val >> 17;
        val ^= val << 5;
        if ( (val & 0xffffff00) == 0 ){
            printf("period %"PRId64", val %x\n", period, val);
        }
        period++;
    } while (val != SEED);

    printf("%"PRId64"\n", period);

}
